packages/core-data/src/queried-data/selectors.js

```diff
@@ -76,12 +77,12 @@ function getQueriedItemsUncached( state, query ) {
 				// This accounts for the fact that queried items are stored by
 				// stable key without an associated fields query. Other requests
 				// may have included fewer fields properties.
-				const field = fields[ f ];
-				if ( ! item.hasOwnProperty( field ) ) {
+				const field = fields[ f ].split( '.' );
```

Comment:
If this value is left as a string, it matches an undefined value in the state— in my testing I had both `meta: { key: 'value' }` and `'meta.key': undefined` on `item`, and `get` matches the string key first, returning undefined. I'm not sure where that other key comes from, but preventing that would be a better fix for this workaround.